src/Sentry.Unity/Il2CppEventProcessor.cs

```diff
@@ -0,0 +1,171 @@
+        private readonly ISentryUnityInfo _sentryUnityInfo;
+        private readonly Il2CppMethods _il2CppMethods;
+
+        public UnityIl2CppEventExceptionProcessor(SentryUnityOptions options, ISentryUnityInfo sentryUnityInfo, Il2CppMethods il2CppMethods)
+        {
+            _options = options;
+            _sentryUnityInfo = sentryUnityInfo;
+            _il2CppMethods = il2CppMethods;
+        }
+
+        public void Process(Exception incomingException, SentryEvent sentryEvent)
+        {
+            _options.DiagnosticLogger?.LogDebug("Running Unity IL2CPP event exception processor on: Event {0}", sentryEvent.EventId);
+
+            var sentryExceptions = sentryEvent.SentryExceptions;
+            if (sentryExceptions == null)
+            {
+                return;
+            }
+            var exceptions = EnumerateChainedExceptions(incomingException);
+
+            // Unity by definition only builds a single library/image,
+            // which we add once to our list of debug images.
+            var debugImages = (sentryEvent.DebugImages ??= new List<DebugImage>());
+            // The il2cpp APIs give us image-relative instruction addresses, not
+            // absolute ones. When processing events via symbolicator, we do want
+            // to have absolute addresses. For this reason, we just add a sentinel
+            // value to the `DebugImage` and `InstructionAddress`, which makes
+            // addresses absolute and still associates the address with the one
+            // and only `GameAssembly` image.
+            var imageAddress = 0x1000;
+            var didAddImage = false;
+
+            foreach (var (sentryException, exception) in sentryExceptions.Zip(exceptions, (se, ex) => (se, ex)))
+            {
+                var sentryStacktrace = sentryException.Stacktrace;
+                if (sentryStacktrace == null)
+                {
+                    // We will only augment an existing stack trace with native
+                    // instructions, so with no stack trace, there is nothing to do
+                    continue;
+                }
+
+                var nativeStackTrace = GetNativeStackTrace(exception);
+
+                if (!didAddImage)
+                {
+                    var codeFile = nativeStackTrace.ImageName;
+                    // NOTE: il2cpp in some circumstances does not return a correct `ImageName`.
+                    // A null/missing `CodeFile` however would lead to a processing error in sentry.
+                    // Since the code file is not strictly necessary for processing, we just fall back to
+                    // a sentinel value here.
+                    if (String.IsNullOrEmpty(codeFile))
+                    {
+                        codeFile = "GameAssembly.fallback";
+                    }
+                    debugImages.Add(new DebugImage
+                    {
+                        Type = _sentryUnityInfo.Platform,
```

Comment:
This needs to be passed down from the script compiled with the game? Isn't it a fixed value?